src/components/SideNavbar.tsx

```diff
@@ -50,30 +51,23 @@ interface navItemProp {
  * @param props
  */
 function NavItem(props: navItemProp) {
-  const history = useHistory();
-
-  /**
-   * click handler than navigates to the new route
-   */
-  function navigate() {
-    history.push(props.route);
-  }
-
   /**
    * returns empty div if the permissions are not met
    */
   return props.requiredPermissions.includes(props.currentPermission) ? (
-    <div
-      onClick={() => {
-        navigate();
-      }}
-      className={`navItem ${
-        props.route === props.currentRoute ? "active" : "inactive"
-      }`}
-    >
-      <div className={`navIcon ${props.name}`}></div>
-      <div className="navTitle">{props.name}</div>
-    </div>
+    <Link to={props.route}>
+      <div
+        onClick={() => {
```

Comment:
Here, `onClick` is unnecessary and the active or inactive class can just be derived from the `useLocation` hook.